addons/binding/org.openhab.binding.evohome/src/main/java/org/openhab/binding/evohome/handler/EvohomeAccountBridgeHandler.java

```diff
@@ -0,0 +1,263 @@
+import org.openhab.binding.evohome.internal.api.models.v2.response.GatewayStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Location;
+import org.openhab.binding.evohome.internal.api.models.v2.response.LocationStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Locations;
+import org.openhab.binding.evohome.internal.api.models.v2.response.LocationsStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.TemperatureControlSystem;
+import org.openhab.binding.evohome.internal.api.models.v2.response.TemperatureControlSystemStatus;
+import org.openhab.binding.evohome.internal.api.models.v2.response.Zone;
+import org.openhab.binding.evohome.internal.api.models.v2.response.ZoneStatus;
+import org.openhab.binding.evohome.internal.configuration.EvohomeAccountConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Provides the bridge for this binding. Controls the authentication sequence.
+ * Manages the scheduler for getting updates from the API and updates the Things it contains.
+ *
+ * @author Jasper van Zuijlen - Initial contribution
+ *
+ */
+public class EvohomeAccountBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(EvohomeAccountBridgeHandler.class);
+    private EvohomeAccountConfiguration configuration;
+    private EvohomeApiClient apiClient;
+    private List<AccountStatusListener> listeners = new CopyOnWriteArrayList<AccountStatusListener>();
+
+    protected ScheduledFuture<?> refreshTask;
+
+    public EvohomeAccountBridgeHandler(Bridge thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        configuration = getConfigAs(EvohomeAccountConfiguration.class);
+
+        if (checkConfig()) {
+            try {
+                apiClient = new EvohomeApiClientV2(configuration);
+
+                // Initialization can take a while, so kick if off on a separate thread
+                scheduler.schedule(new Runnable() {
+                    @Override
+                    public void run() {
+                        if (apiClient.login()) {
+                            if (checkInstallationInfoHasDuplicateIds(apiClient.getInstallationInfo())) {
+                                startRefreshTask();
+                            } else {
+                                updateAccountStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                        "System Information Sanity Check failed");
+                            }
+                        } else {
+                            updateAccountStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                    "Authentication failed");
+                        }
+                    }
+                }, 0, TimeUnit.SECONDS);
+            } catch (Exception e) {
+                logger.error("Could not start API client", e);
```

Comment:
Is it unexpected to ever get here, so that it is a bug that needs to be reported? Then logging ERROR is ok, but you might also directly want to throw an exception to signal that your handler initialisation failed.
No need to set the Thing status in that case.

Reply:
Yes, that is true. Does that mean that I can remove the try...catch altogether?